### PR TITLE
Per attachment scoping transforms

### PR DESCRIPTION
**Pull request by juanmuscaria**
Effectively replaces #211

### CODE REVIEW

File: crafting-dead-core/src/main/java/com/craftingdead/core/client/renderer/item/GunRenderer.java [hunk 1]
```diff
@@ -447,7 +449,10 @@ private void renderGun(
         ? TransformationHelper.slerp(normalTransform,
             gun.hasIronSight()
                 ? this.properties.aimTransform()
-                : this.properties.scopeAimTransform(),
+                : this.properties.scopeAimTransform().getOrDefault(gun.getAttachments().stream()
```
Comment (josephtarbit) [inline crafting-dead-core/src/main/java/com/craftingdead/core/client/renderer/item/GunRenderer.java]: This should really be cached somehow, creating a stream every frame isn't the nicest.

**Review by josephtarbit — CHANGES_REQUESTED** [review]

File: crafting-dead-core/src/main/java/com/craftingdead/core/client/renderer/item/GunRenderer.java (hunk 1 above)
Comment (juanmuscaria) [reply crafting-dead-core/src/main/java/com/craftingdead/core/client/renderer/item/GunRenderer.java]: Migrated the attachments to a map, removing the need for a stream

File: crafting-dead-core/src/main/java/com/craftingdead/core/world/item/gun/AbstractGun.java [hunk 2]
```diff
@@ -786,7 +790,8 @@ public void deserializeNBT(CompoundTag tag) {
         .map(Tag::getAsString)
         .map(ResourceLocation::new)
         .map(Attachments.registry.get()::getValue)
-        .collect(Collectors.toSet()));
+        .filter(Objects::nonNull)
```
Comment (josephtarbit) [inline crafting-dead-core/src/main/java/com/craftingdead/core/world/item/gun/AbstractGun.java]: Shouldn't be necessary.

File: crafting-dead-core/src/main/java/com/craftingdead/core/world/item/gun/AbstractGun.java [hunk 3]
```diff
@@ -759,8 +762,9 @@ public CompoundTag serializeNBT() {
     }
     tag.putString("ammoProviderType", this.ammoProvider.getType().getRegistryName().toString());
     tag.put("ammoProvider", this.ammoProvider.serializeNBT());
-    var attachmentsTag = this.attachments.stream()
+    var attachmentsTag = this.attachments.values().stream()
         .map(Attachment::getRegistryName)
+        .filter(Objects::nonNull) // Is this? it removes the null warning - juanmuscaria
```
Comment (josephtarbit) [inline crafting-dead-core/src/main/java/com/craftingdead/core/world/item/gun/AbstractGun.java]: Shouldn't be necessary.

**Review by josephtarbit — CHANGES_REQUESTED** [review]

**Pull request merged**